### PR TITLE
chore: pydantic and mypy tweaks

### PR DESCRIPTION
**Pull request by PrettyWood**

### CODE REVIEW

File: server/src/weaverbird/pipeline/conditions.py [hunk 1]
```diff
@@ -55,8 +55,8 @@ class DateBoundCondition(BaseModel):
 
 
 class BaseConditionCombo(BaseCondition, ABC):
-    class Config(PopulatedWithFieldnames):
-        ...
+    class Config(BaseConfig):
+        allow_population_by_field_name = True
```
Comment (Sanix-Darker) [inline server/src/weaverbird/pipeline/conditions.py:59]: ah 👍🏽 !
I see now, thanks for this!

File: server/src/weaverbird/pipeline/steps/ifthenelse.py [hunk 2]
```diff
@@ -22,13 +22,13 @@ class Config(PopulatedWithFieldnames):
...
 
 class IfthenelseStep(BaseStep, IfThenElse):
-    class Config(PopulatedWithFieldnames):
-        ...
-
     name: Literal["ifthenelse"] = "ifthenelse"
     new_column: ColumnName = Field(alias="newColumn")
 
+    class Config(BaseConfig):
+        allow_population_by_field_name = True
```
Comment (lukapeschke) [inline server/src/weaverbird/pipeline/steps/ifthenelse.py:29]: Is this needed ? It's in boith `BaseStep` and `IfthenElse`. Or does the multiple inheritance F up pydantic's config ?

File: server/src/weaverbird/pipeline/steps/ifthenelse.py [hunk 3]
```diff
@@ -22,13 +22,13 @@ class Config(PopulatedWithFieldnames):
...
+
 
 class IfThenElseStepWithVariables(IfthenelseStep, StepWithVariablesMixin):
-    class Config(PopulatedWithFieldnames):
-        ...
+    class Config(BaseConfig):
+        allow_population_by_field_name = True
```
Comment (lukapeschke) [inline server/src/weaverbird/pipeline/steps/ifthenelse.py:34]: Same remark here, It's already in the parent

**Review by lukapeschke — APPROVED** [review]
LGTM apart from the two inline comments. Thanks!

**Review by Sanix-Darker — APPROVED** [review]

File: server/src/weaverbird/pipeline/steps/ifthenelse.py (hunk 2 above)
Comment (PrettyWood) [reply server/src/weaverbird/pipeline/steps/ifthenelse.py:29]: I know but it fucks up mypy ;)

File: server/src/weaverbird/pipeline/steps/ifthenelse.py (hunk 3 above)
Comment (PrettyWood) [reply server/src/weaverbird/pipeline/steps/ifthenelse.py:34]: Same remark here as well :P

**Pull request merged**